wp-content/themes/core/blocks/core/posttemplate/style.pcss

```
@@ -0,0 +1,46 @@
+/**
+ * Styles specific to this block
+ */
+
+/**
+ * Vertical (list layout) post template spacing
+ */
+.wp-block-post-template:not(.is-layout-grid) .wp-block-post + .wp-block-post {
+	margin-block-start: var(--spacer-40);
+}
+
+/**
+ * Horizontal (grid layout) post template spacing / media queries
+ *
+ * Essentially the calculations take the amount of columns and subtract
+ * the amount of spacers divided by the amount of columns. This gets the column
+ * width. The calculations subtract an extra pixel due to calc not enjoying our
+ * use of clamped spacer variables.
+ */
+.wp-block-post-template.is-layout-grid {
+	gap: var(--spacer-40);
+	grid-template-columns: 1fr;
+
+	@media (--mq-wp-medium) {
+		--post-template-column-count: 2;
+		grid-template-columns: repeat(var(--post-template-column-count), minmax(0, 1fr));
+	}
+
+	&.columns-3 {
+
+		@media (--mq-wp-medium) {
+			--post-template-column-count: 1;
+		}
+
+		@media (--viewport-full) {
+			--post-template-column-count: 3;
+		}
+	}
+
+	&.columns-4 {
+
+		@media (--viewport-full) {
+			--post-template-column-count: 4;
+		}
+	}
+}
```

Just spitballing: I wonder if this is any more readible/better?

```suggestion
.wp-block-post-template.is-layout-grid {
	--post-template-grid-template-columns: 1fr;
	gap: var(--spacer-40);
	grid-template-columns: var(--post-template-grid-template-columns);
	
	&.columns-2,
	&.columns-4 {

		@media (--mq-wp-medium) {
			--post-template-grid-template-columns: repeat(2, minmax(0, 1fr));
		}
	}

	&.columns-3 {

		@media (--viewport-full) {
			--post-template-grid-template-columns: repeat(3, minmax(0, 1fr));
		}
	}

	&.columns-4 {

		@media (--viewport-full) {
			--post-template-grid-template-columns: repeat(4, minmax(0, 1fr));
		}
	}	
}
```